### DISCUSSION TITLE
FSP Install Issue?

### DISCUSSION THREAD

**Discussion by AlecKon** [retronas/discussions#19] [Q&A] [answered]
Not very experienced with linux beyond following instructions. Have a retronas setup on a SurfacePro 5 running latest debian 11. 

All services install fine and are working except FSP which I was interested in trying out. 
Install fails. Folders exist (created by ROM import - I think).

TASK [Generic ROM dir - build systems directory layout] **************************************************************************************
Monday 27 March 2023  11:20:24 +1100 (0:00:00.065)       0:00:02.253 ********** 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "ln -sf /storage/roms/nec/pcengine /storage/roms/nec/turbografx16;ln -sf /storage/roms/nec/pcengine/iso /storage/roms/nec/turbografxcd;ln -sf /storage/roms/nintendo/famicom /storage/roms/nintendo/nes;ln -sf /storage/roms/nintendo/superfamicom /storage/roms/nintendo/snes;ln -sf /storage/roms/sega/megacd /storage/roms/sega/segacd;ln -sf /storage/roms/sega/megadrive /storage/roms/sega/genesis;", "delta": "0:00:00.004525", "end": "2023-03-27 11:20:25.106618", "msg": "non-zero return code", "rc": 1, "start": "2023-03-27 11:20:25.102093", "stderr": "ln: failed to create symbolic link '/storage/roms/nec/turbografx16': Operation not permitted\nln: failed to create symbolic link '/storage/roms/nec/turbografxcd': Operation not permitted\nln: failed to create symbolic link '/storage/roms/nintendo/nes': Operation not permitted\nln: failed to create symbolic link '/storage/roms/nintendo/snes': Operation not permitted\nln: failed to create symbolic link '/storage/roms/sega/segacd': Operation not permitted\nln: failed to create symbolic link '/storage/roms/sega/genesis': Operation not permitted", "stderr_lines": ["ln: failed to create symbolic link '/storage/roms/nec/turbografx16': Operation not permitted", "ln: failed to create symbolic link '/storage/roms/nec/turbografxcd': Operation not permitted", "ln: failed to create symbolic link '/storage/roms/nintendo/nes': Operation not permitted", "ln: failed to create symbolic link '/storage/roms/nintendo/snes': Operation not permitted", "ln: failed to create symbolic link '/storage/roms/sega/segacd': Operation not permitted", "ln: failed to create symbolic link '/storage/roms/sega/genesis': Operation not permitted"], "stdout": "", "stdout_lines": []}

Or is this ignorable...as folders already exist?
have tried running retronas as sudo

**Comment by sairuk** [comment, marked as answer]
what kind of mount is `/storage` ? samba/cifs?

**Reply by AlecKon** [reply]
exfat filesystem. 
hmm. thought it was ext4.

**Reply by sairuk** [reply]
for extfat or any ms style you'll need `mfsymlinks` mount option for symlinks to have a chance of working, otherwise if you can convert it to ext4 for use or other fs that support symlinks

**Reply by AlecKon** [reply]
Thanks...definitely a user error using an exfat format. Investigated mfsymlinks option...but I dont really know what I am doing and likely to break something else :)
Will format ext4. 
Thanks for the guidance.
